### DISCUSSION TITLE
How to get rid of csrf issues

### DISCUSSION THREAD

**Discussion by amolvishwakarma** [pomerium/pomerium#2327] [Q&A]
Below is my sample configuration file: 

# See detailed configuration settings : https://www.pomerium.io/docs/reference/reference/

# this is the domain the identity provider will callback after a user authenticates
pomerium_debug: true

authenticate_service_url: https://example.com:9443
authenticate_callback_path: /oauth2/callback

# certificate settings:  https://www.pomerium.io/docs/reference/certificates.html
autocert: false
certificate_file: "fullchain.pem"
certificate_key_file: "privkey.pem"
# REMOVE FOR PRODUCTION
autocert_use_staging: true

# identity provider settings : https://www.pomerium.io/docs/identity-providers.html
idp_provider: google
idp_provider_url: https://accounts.google.com
idp_client_id: example.apps.googleusercontent.com
idp_client_secret: examplesecrets

# Generate 256 bit random keys  e.g. `head -c32 /dev/urandom | base64`
cookie_domain: example.com
cookie_secret: somerandomsecrets

# https://www.pomerium.io/configuration/#policy
policy:
  - from: https://example.com:9443
    to: https://example.com:10443
    allowed_domains:
      - gmail.com
    pass_identity_headers: true

**Comment by calebdoxsey** [comment]
Hi @amolvishwakarma ,

Do you have more details about the CSRF issue you're seeing?

**Reply by amolvishwakarma** [reply]
Csrf issue is fixed now but I am getting error: 421 misredirected request is it due to the same subdomain I am using everywhere but port is different

**Reply by calebdoxsey** [reply]
We don't support binding multiple ports. Are you going through another proxy before hitting pomerium? ie `http://example.com:9443 -> pomerium:443` and `http://example:9444 -> pomerium:443` through some sort of TCP proxy? I don't know if I've ever tried to do that before.

**Reply by amolvishwakarma** [reply]
I am using docker where internal port is 443 and external port is 9443 and 10443

**Reply by amolvishwakarma** [reply]
No any other proxy is running

**Reply by calebdoxsey** [reply]
ok. I'm not sure. Do you want to open a new issue following the issue template?

**Reply by amolvishwakarma** [reply]
Yes
